\documentclass{article}
\usepackage{arxiv}
\usepackage[utf8]{inputenc} % allow utf-8 input
\usepackage[T1]{fontenc}    % use 8-bit T1 fonts
\usepackage[hidelinks]{hyperref}       % hyperlinks
\usepackage{url}            % simple URL typesetting
\usepackage{amsfonts}       % blackboard math symbols
\usepackage{microtype}      % microtypography
\usepackage{graphicx}
\usepackage{doi}
\usepackage{authblk}
\usepackage{siunitx}

\graphicspath{ {./figures/} }


\usepackage[backend=biber,sorting=nyt,style=authoryear, maxcitenames=2, maxbibnames=10, giveninits=true, terseinits=true, date=year, uniquename = init ]{biblatex}
\addbibresource{GR_2020_references.bib}
\renewbibmacro{in:}{}
\renewcommand*{\revsdnamepunct}{}
\DeclareNameAlias{sortname}{family-given}
\DefineBibliographyExtras{english}{%
  \renewcommand*{\finalnamedelim}{\addcomma\addspace}%
}
\DeclareDelimFormat[bib]{nameyeardelim}{%
\addperiod\space}
\DeclareFieldFormat[article, inbook, incollection, inproceedings, misc, thesis, unpublished]{title}{#1}

\usepackage{xpatch,filecontents}

\begin{filecontents}{\jobname.bib}
@misc{A01,author={Author, Anne},year={2001},title={Alpha}}
@misc{A02,author={Author, Anne},year={2001},title={Beta}}
\end{filecontents}
\addbibresource{\jobname.bib}

\xpatchbibmacro{date+extradate}{%
  \printtext[parens]%
}{%
  \setunit*{\addperiod\space}%
  \printtext%
}{}{} 

\renewbibmacro*{journal}{%
  \iffieldundef{shortjournal}
    {%
      \iffieldundef{journaltitle}
        {}
        {%
          \printtext[journaltitle]
            {%
              \printfield[titlecase]{journaltitle}%
              \setunit{\subtitlepunct}%
              \printfield[titlecase]{journalsubtitle}%
             }%
         }%
    }
    {\printtext[journaltitle]{\printfield[titlecase]{shortjournal}}}%
}
\DeclareFieldFormat[article]{volume}{\textbf{#1}\addcolon\space}
\DeclareFieldFormat{pages}{#1}
\AtEveryBibitem{\clearfield{number}}
 
\title{Glucocorticoid receptor collaborates with pioneer factors and AP-1 to execute genome-wide regulation}

\author[1]{Erin M. Wissink}
\author[2]{Delsy M. Martinez}
\author[2]{Kirk T. Ehmsen}
\author[2]{Keith R. Yamamoto}
\author[1, 3]{John T. Lis}

\affil[1]{Department of Molecular Biology and Genetics, Cornell University, Ithaca, NY}
\affil[2]{Department of Cellular and Molecular Pharmacology, University of California San Francisco School of Medicine, San Francisco, CA}
\affil[3]{Corresponding author, email: jtl10@cornell.edu}

\renewcommand{\shorttitle}{Glucocorticoid receptor collaborates with pioneer factors and AP-1}



\begin{document}
{\fontfamily{phv}\selectfont
\maketitle

\begin{abstract}
The glucocorticoid receptor (GR) regulates transcription through binding to specific DNA motifs, particularly at enhancers. While the motif to which it binds is constant across cell types, GR has cell type-specific binding at genomic loci, resulting in expression of different genes. The presence of other bound transcription factors (TFs) is hypothesized to strongly influence where GR binds. Here, we addressed the roles of other TFs in the glucocorticoid response by comparing changes in GR binding and nascent transcription at promoters and distal candidate \emph{cis}-regulatory elements (CCREs) in two distinct human cancer cell types. We found that after glucocorticoid treatment, GR binds to thousands of loci that are primarily distal to promoters. The majority of these binding sites are cell-type specific, and they are associated with pioneer factor binding. Transcription correlates with active enhancers, yet only a small fraction of GR occupied regions (GORs) had a concurrent increase in transcription. GORs with increased transcription were specifically enriched for AP-1 binding prior to glucocorticoid treatment. These results support a model of transcriptional regulation in which multiple classes of TFs are required. The pioneer factors increase chromatin accessibility, facilitating the binding of GR and additional factors. AP-1 binding poises a fraction of accessible sites to be rapidly transcribed upon glucocorticoid-induced GR binding. The coordinated activity of multiple TFs then results in cell type-specific changes in gene expression. We anticipate that many models of inducible gene expression also require multiple distinct TFs that act at multiple steps of transcriptional regulation.
\end{abstract}


\section*{Introduction}
Transcription of gene networks can be regulated by a variety of extracellular stimuli, including developmental signals, hormones, and environmental changes. This regulation is specified by DNA sequence elements associated with promoters and enhancers (\cite{zabidi_regulatory_2016, long_ever-changing_2016}). These regulatory elements bind to sequence-specific transcription factors (TFs) that can alter DNA accessibility, bring in co-activators, and recruit the pre-initiation complex, including RNA Polymerase II (Pol II) (\cite{core_promoter-proximal_2019}). While gene promoters are required for transcription, they are often not sufficient and instead additionally require the activity of distal enhancers (\cite{zabidi_regulatory_2016}).

Enhancers are sequences that regulate target promoters by acting as platforms for TF binding (\cite{zabidi_regulatory_2016, long_ever-changing_2016}). These elements are located within or between genes, do not necessarily regulate the nearest promoter, and can associate with candidate target promoters via direct looping (\cite{schoenfelder_long-range_2019}), which enables enhancers to be located up to a megabase from their target (\cite{lettice_disruption_2002}). While genes are frequently transcribed in many cell types, enhancers generally have more cell-type specific activity because they bind TFs or other co-factors that have restricted expression (\cite{andersson_atlas_2014, weikum_glucocorticoid_2017}). Enhancer activity correlates with chromatin accessibility and histone modifications, particularly H3K27ac and more H3K4me1 than H3K4me3 (\cite{heintzman_distinct_2007, ernst_discovery_2010}). Assessing which enhancers have functional activity is an area of open investigation. One model predicts that accessible sites with H3K27ac that are in contact with promoters are active (\cite{fulco_activity-by-contact_2019}). Others have shown that those distal sites that are transcribed as enhancer RNAs (eRNAs) (\cite{ashe_intergenic_1997, de_santa_large_2010, kim_widespread_2010, wang_reprogramming_2011}) have greater functional activity in reporter assays than non-transcribed sites (\cite{henriques_widespread_2018, mikhaylichenko_degree_2018, wissink_nascent_2019}). These eRNAs can be detected with high sensitivity by sequencing nascent RNAs because they produce bidirectional transcripts, similar to promoters (\cite{andersson_atlas_2014, core_analysis_2014, scruggs_bidirectional_2015, mikhaylichenko_degree_2018, wissink_nascent_2019}). The architecture of promoters and enhancers are similar, as TFs bind in the nucleosome-depleted region between the two core promoters for both types of elements (\cite{core_analysis_2014, andersson_unified_2015, tome_single-molecule_2018, tippens_transcription_2020}), and the specific TFs bound likely determine which elements have enhancer or promoter activity (\cite{andersson_determinants_2020}). 

The TFs that bind to regulatory elements play different roles, and the constellation of TFs that can bind determine the stimuli to which enhancers respond (\cite{weikum_glucocorticoid_2017, vihervaara_molecular_2018}). An individual enhancer can bind to multiple TFs, including ones that bind to closed chromatin (“pioneer factors”), chromatin remodelers, and factors that release Pol II from promoter-proximal pausing (\cite{andersson_determinants_2020}). The well-studied heat shock system demonstrates the importance of multiple factors acting at different steps of transcription. HSF1 activates transcription of hundreds of genes by binding to the DNA motif known as the heat shock element, then releasing paused polymerases into active elongation (\cite{bunch_trim28_2014, mahat_mammalian_2016}).  However, less than half of the HSF1-bound promoters have increased transcription after heat shock (\cite{mahat_mammalian_2016}), demonstrating that this TF is not sufficient for transcription. Instead, activation of genes by HSF1 relies on pre-binding of other factors (\cite{duarte_transcription_2016}), accessible chromatin (\cite{vihervaara_transcriptional_2017}), and pre-formed enhancer-promoter loops (\cite{ray_chromatin_2019}). Similarly, signaling from inflammation uses pre-wired looping (\cite{jin_high-resolution_2013}) and pre-binding of a pioneer factor (\cite{escoubet-lozach_mechanisms_2011}). Thus, studies of stress-specific transcription networks have uncovered principles of how other factors impact gene regulation.

In the current work, we investigated transcriptional regulation by the glucocorticoid receptor (GR). Like HSF1, GR binds to a DNA motif, in this case the GR binding sequences (GBSs), in response to an extracellular stimulus, namely, glucocorticoid hormones. These ligands bind to cytosolic GR, triggering its release from chaperones, and allowing it to translocate to the nucleus to bind GBSs (\cite{miranda_complex_2013, meijsing_mechanisms_2015, weikum_glucocorticoid_2017}). GR recognizes a cell type-specific subset of GBS motifs in different cell types, resulting in regulation of cell type-specific target genes (\cite{so_determinants_2007, john_chromatin_2011, love_role_2017, thormann_genomic_2018, franco_immune_2019}). GR mostly binds to accessible chromatin (\cite{john_chromatin_2011, grontved_cebp_2013, love_role_2017}) that is pre-marked with H3K27ac and EP300 binding (\cite{mcdowell_glucocorticoid_2018}) at sites that are typically >10 kb from promoters (\cite{so_determinants_2007, reddy_genomic_2009}). GR binds to roughly 104 specific genomic loci in a single cell type, yet only a small fraction of these GR-occupied regions (GORs) are thought to have enhancer activity (\cite{vockley_direct_2016}). While studies have interrogated the chromatin features and factors that allow GR binding, how these are coordinated to create a functional enhancer remains unclear.

Here, we are utilizing two cell types with different lineages to disentangle the roles of different TF classes in the GR response. These cell lines are A549, a lung adenocarcinoma, and U2OS, an osteosarcoma. With nascent RNA sequencing, we determined which genes and transcribed putative enhancers are commonly and specifically regulated in response to GR signaling in these cells. The transcribed putative enhancers comprise only a small fraction of the GORs in each cell type, so we identified TF motifs that are associated with transcription. Finally, we compared the cell type-specific factors that are enriched at cell type-specific GORs. With these data, we demonstrate the importance of the binding of TFs that potentially act in different roles, including making chromatin more accessible, enabling the activity of GR, and activating transcription in the cellular response to glucocorticoids, and we discuss the features common to other transcriptionally inducible systems. 



\section*{Results}
\label{sec:headings}

\subsection*{GR signaling has a stronger transcriptional effect in U2OS cells than A549 cells}

We investigated GR’s transcriptional regulatory activity in two cell lines, a lung carcinoma (A549) and an osteosarcoma (U2OS).  A549 cells have endogenous GR expression, whereas these U2OS cells have transgenic GR expression (\cite{rogatsky_glucocorticoid_1997}).  To examine the primary effects of GR on the nascent transcriptome, we treated these cells with either ethanol (control) or the synthetic corticosteroid dexamethasone (dex) for 45 minutes, followed by PRO-seq (Figure \ref{fig:fig1}A). Increases in nascent transcription are readily apparent at this time point, as seen at the \emph{FKBP5} gene, which is induced by dex in both cell types (Figure \ref{fig:fig1}B). Both cell types have genes that are up- or downregulated after dex treatment, although far more genes are differentially expressed after dex treatment in U2OS than in A549 cells (Figure \ref{fig:fig1}C-E, Figure S1A,B, Table S1), showing that the U2OS cells are more responsive to GR signaling. The changes in gene transcription correlate significantly between the two cell types (Figure \ref{fig:fig1}E). While 450 genes that are differentially expressed in A549 cells show a concordant and significant change in expression in U2OS cells, 322 genes show A549-specific expression changes, and 2011 have U2OS-specific effects (Figure \ref{fig:fig1}F).

\begin{figure}[ht!]
\includegraphics[width=\textwidth]{figures/Figure1_gene_expression_2021-04.png}
\caption{\textbf{Genic transcription changes after glucocorticoid stimulation. (A)} Experimental set-up. \textbf{(B)} Browser shot depicting nascent RNA from PRO-seq reads for the beginning of the \emph{FKBP5} gene in A549 (left) and U2OS (right) cells. Coordinates are chr6:35,634,000-35,690,000 (hg38). \textbf{(C, D)} MA plots for global gene expression changes in A549 \textbf{(C)} and U2OS \textbf{(D)} cells, comparing treatment with ethanol (vehicle control, EtOH) to treatment with 1 nM dex (upper) or 100 nM dex (lower). Numbers of significantly differentially expressed genes are denoted; Benjamini-adjusted p < 0.05. \textbf{(E)} Comparison of fold-change differences in expression for genes called as significantly different in A549 and/or U2OS cells, after treatment with 1 nM dex (upper) or 100 nM dex (lower). Spearman correlation coefficient is displayed; p < \num{2.2e-16} for both. \textbf{(F)} Expression change heat-map for genes in panel E. Genes have been categorized by their response to 100 nM dex treatment.}
\label{fig:fig1}
\end{figure}

A previous study showed that some mature mRNAs are significantly downregulated after 3 hours of treatment with 1 nM dex but upregulated with 100 nM dex, or vice versa (\cite{chen_incoherent_2013}). Dose-dependent gene expression changes are not apparent in the nascent transcriptome after 45 minutes dex treatment (Figure S1C,D), suggesting this incoherent feed-forward loop occurs later in signaling and/or post-transcriptionally. 

One possible explanation for the differences in GR responsiveness is that U2OS cells have ~5 times more GR protein than A549 cells (\cite{chen_incoherent_2013}), yet the fact that A549 and U2OS cells each have unique target genes suggest that other factors are also playing a critical role. Some genes are inversely dex-responsive in the two cell types tested (i.e., activated in one and repressed in the other) (Figure \ref{fig:fig1}E,F), further indicating that GR is not the sole determinant of the cellular response to dex. 

\subsection*{GR binds different sites in A549 and U2OS cells}

To identify underlying causes of cell-type specific gene expression changes after dex treatment, we performed ChIP-seq to GR in A549 and U2OS cells after a 90 min treatment with 100 nM dex. We found thousands of GR-occupied regions (GORs) across the genome in each cell type, with more widespread occupancy in U2OS cells (Figure \ref{fig:fig2}A, Table S2). Each cell type has specific binding sites, yet there are 4596 that are shared (63\% for A549, 18\% for U2OS). We compared our ChIP-seq data to published ATAC-seq data (\cite{encode_project_consortium_integrated_2012, davis_encyclopedia_2018, lee_different_2018}), and found that GR binds to both highly and lowly accessible chromatin in both A549 and U2OS cells (Figure \ref{fig:fig2}A).

\begin{figure}[hb!]
\includegraphics[width=\textwidth]{figures/Figure2_v2_2021-04.png}
\caption{\textbf{Cell type determinants of GR binding. (A)} GR-occupied regions (GORs) were identified from GR ChIP-seq signal and categorized as being shared between A549 cells, specific to A549 cells, or specific to U2OS cells. GR ChIP-seq signal is shown in control cells and cells treated with 100 nM dex; ATAC-seq signal is shown in control cells. Signal is -log10(p-value) of enrichment. \textbf{(B)} In A549 cells, metaplots of mean signal for GR ChIP-seq in 100 nM dex-treated cells, and ATAC-seq and ChIP-seq data for CEBPB, FOXA2, and FOSL2 in control cells are shown for shared GORs (blue), A549-specific GORs (green), and U2OS-specific GORs (pink). Signal is the -log10 p-value of enrichment, and shading indicates the standard deviation. \textbf{(C)} The log-odds scores for the GR binding sequence (GBS) were found for GORs that are shared or cell-type specific. The score of the best GBS for each is plotted. * p < \num{1e-8}. \textbf{(D)} Genomic distribution of GORs.}
\label{fig:fig2}
\end{figure}

The cell type-specific binding of GR demonstrates that information beyond the GBS directs its binding, so we used motif searching to identify other factors that may associate with GORs. Our search was limited to +/- 150 bp from the summit of GR binding, as this distance captures the regulatory information between two core promoters that are associated with candidate enhancers (\cite{core_analysis_2014, scruggs_bidirectional_2015, tippens_transcription_2020}). Motifs for FOXA, CEBP, and AP-1 (which consists of a heterodimer of FOS, JUN, and ATF family members) were enriched in A549-specific GORs, as compared to U2OS-specific GORs (Table S3). Additionally, expression of FOXA1, FOXA2, CEBPA, and CEBPB are significantly upregulated in A549 cells, as compared to U2OS cells, whereas some but not all members of the AP-1 complex have higher A549 expression (Figure S2A). Each of these factors have reported pioneer activity (\cite{lupien_foxa1_2008, belikov_foxa1_2009, biddie_transcription_2011, grontved_cebp_2013}) and have been implicated in affecting GR binding (\cite{so_determinants_2007, belikov_foxa1_2009, biddie_transcription_2011, grontved_cebp_2013, starick_chip-exo_2015, vockley_direct_2016, mcdowell_glucocorticoid_2018, hoffman_brg1_2018}). The higher relative expression of FOXA and CEBP in A549 cells may increase chromatin accessibility at A549-specific GORs, enabling GR binding. 

One highly significantly enriched motif at U2OS-specific GORs, as compared to A549-specific GORs, is the RUNX motif (Table S3). The RUNX family of transcription factors have pioneering activity \cite{wang_transcription_2018, veeken_natural_2019, lee_runx3_2019}) and may act in higher-order chromatin structure (\cite{barutcu_runx1_2016}). RUNX2 is required for osteoblast differentiation, is expressed in U2OS cells (\cite{komori_regulation_2005, lucero_cancer-related_2013}), and has significantly higher expression in U2OS than A549 cells (Figure S2A). Other motifs include the GBS (GCR) and the nearly identical progesterone receptor motif (PRGR) (Table S3). While the GBS enrichment is significant, the GBS is found at high fraction of both U2OS- and A549-specific GORs (96.5\% and 94\% respectively). U2OS GR binding may be enabled by the higher expression of RUNX2, which could increase chromatin accessibility at U2OS-specific GORs.

Protein-binding data is more informative than the presence of enriched motifs. Unfortunately, little publicly available ChIP-seq data for U2OS cells exist; however, ChIP-seq data have been published for FOXA2, CEBPB, FOSL2, JUN, and JUNB in A549 cells (\cite{donaghey_genetic_2018, mcdowell_glucocorticoid_2018}). We compared the presence of these factors at GORs that are either shared or cell type-specific in cells that were not treated with dex. We also compared chromatin accessibility by ATAC-seq  prior to dex treatment (\cite{encode_project_consortium_integrated_2012, davis_encyclopedia_2018, lee_different_2018}). Together, the binding of these factors and ATAC-seq reveal the chromatin landscape available to GR when its binding is rapidly activated by dex treatment (Figure \ref{fig:fig2}B, Figure S2B). A549-specific GORs are overall the most accessible prior to dex addition, and they have the strongest binding of CEBPB, FOXA2, and AP-1 components (Figure \ref{fig:fig2}B, Figure S2B). Sites that are specific to U2OS cells display the least accessibility and are not enriched for motifs for the A549-enriched components: CEBPB, FOXA2, and AP-1 (Figure \ref{fig:fig2}B, Figure S2B). While shared GORs have less binding of these pioneer factors, they have the greatest magnitude of GR binding and higher AP-1 binding than U2OS-specific sites. These results suggest that pioneer factor-binding prior to dex treatment promotes chromatin accessibility and thereby allows GR binding to cell type-specific sites. We predict that in U2OS cells, binding of RUNX2 guides GR binding. The higher GR binding at GORs that are shared between U2OS and A549 cells suggests these sites rely on other TFs, genomic features, and/or the quality of the GBS motif. 

To examine the impact of GBS strength, we found the log odds scores of every GBS at a GOR (Figure \ref{fig:fig2}C). Shared GORs have the strongest GBS, followed by U2OS GORs, then A549 GORs. Similarly, estrogen receptor (ER) binding sites that are shared between two cell types have stronger ER elements than cell-type specific ones (\cite{gertz_distinct_2013}). Further, GORs that are specific to A549 cells have significantly fewer GBS elements per binding site than U2OS-specific or shared sites (Figure S2C). GORs that are shared between cell types may not rely on the prior binding of a cell type-specific pioneer factor. Instead, the ubiquitously expressed AP-1 may bind and open chromatin, and then multiple GBS elements that strongly bind to GR could permit GR to destabilize nucleosome binding to chromatin.

We interrogated the genomic loci to which GR binds (Figure \ref{fig:fig2}D) and considered a promoter to be within 1 kb from an annotated TSS. Nearly all binding is to intra- and intergenic sites, which may be enhancers. We compared transcription changes at genes with and without promoter binding of GR. GR binding to promoters is significantly associated with gene induction in both cell types, but not with repression (Figure S2D). Genes with GR binding are significantly upregulated after dex addition, as compared to genes without binding, in both cell types (Figure S2E). Promoter binding of GR therefore does not generally mediate transcriptional repression at this time point.

\subsection*{Enhancers have cell type-specific transcriptional changes in response to GR signaling}

The expression and activity of enhancers are largely cell-type specific (\cite{andersson_atlas_2014}), and transcribed eRNAs can be readily captured by PRO-seq (\cite{wissink_nascent_2019}). Here, we used the program dREG to identify sites of bidirectional transcription (\cite{wang_identification_2019, chu_discovering_2019}) from our PRO-seq data, then called loci that were at least 1 kb from an annotated transcription start site as putative enhancers (candidate \emph{cis}-regulatory elements, CCREs) (\cite{moore_expanded_2020}). Of the 32,237 bidirectional transcripts detected in A549 and U2OS cells, 13,522 overlapped promoters, while the 11,207 intergenic and 7508 intragenic loci were considered CCREs (Figure S3A). The promoter-matching transcripts were predominantly shared between cell types, whereas CCREs were more likely to be cell type-specific (Figure S3A). We found that 5\% of CCREs exhibit a significant change in transcription after dex treatment (dex-responsive CCREs; Table S4). As seen in the genic transcription, dex-responsive CCREs are largely cell type-specific, with a larger dex response in U2OS than in A549 cells (Figure \ref{fig:fig3}). Examples of CCREs are shown (Figure \ref{fig:fig3}A), including one that is dex-inducible in both cell types (left), in only A549 cells (center), and in only U2OS cells (right). GR binding, as measured by ChIP-seq after 100 nM dex treatment, is also included. In one of the example CCREs, GR binding is present in both cell types, yet changes in transcription are present only in A549 cells (Figure 3A, center). As discussed in more depth below, binding of GR is not sufficient to predict which CCREs will be differentially expressed in each cell type.

The global trends for dex-responsive transcription at CCREs are similar to genes. A subset of CCREs have the same response in both cell types, but overall U2OS cells have greater changes in transcription at both 1 nM and 100 nM dex treatment (Figures \ref{fig:fig3}B-E, S2B-D). The stronger U2OS response is apparent after the 1 nM treatment, where only two CCREs have significant differential expression in A549 cells, as compared to 411 CCREs in U2OS, and the changes between cell types have a smaller correlation than after 100 nM dex treatment (Figure \ref{fig:fig3}B-E). While there are similar numbers of up- and down-regulated genes, CCREs are more likely to exhibit induced rather than repressed transcription after dex treatment in both cell types (Figures \ref{fig:fig3}B-E).


\subsection*{CCREs showing changes in PRO-seq signal have characteristics of GR-induced enhancers}

\begin{figure}[ht!]
\includegraphics[height=7in]{figures/Figure3_v2.png}
\caption{\textbf{Candidate \emph{cis}-regulatory element (CCRE) transcription changes after glucocorticoid stimulation. (A)} Browser shot depicting nascent RNA from PRO-seq (upper three panels) and GR ChIP-seq (lower) in A549 and U2OS cells. Included are a CCRE induced in both cell types (left), only in A549 cells (center), and only in U2OS cells (right). In PRO-seq, red and blue indicate sense and anti-sense transcription, respectively. Dashed boxes delimit CCREs determined by dREG. \textbf{(B, C)} MA plots for global CCRE expression changes in A549 \textbf{(B)} and U2OS \textbf{(C)} cells, comparing treatment with EtOH to treatment with 1 nM dex (upper) or 100 nM dex (lower). Numbers of significantly differentially expressed genes are shown; Benjamini-adjusted p < 0.05. \textbf{(D)} Comparison of fold-change differences in expression for CCREs called as significantly different in A549 and/or U2OS cells, after treatment with 1 nM dex (upper) or 100 nM dex (lower). Spearman correlation coefficient is displayed; p < \num{2.2e-16} for both. \textbf{(E)} Expression change heat-map for CCREs in panel \textbf{D}. CCREs have been categorized by their response to 100 nM dex treatment. One CCRE with opposing responses in A549 and U2OS cells was omitted from the heatmap.}
\label{fig:fig3}
\end{figure}

Our identification of dex-responsive CCREs was based solely on PRO-seq data, and thus did not incorporate information from proteins binding to DNA, transcription factor motifs, histone modifications, or chromatin accessibility. We therefore looked for expected features of GR-responsive enhancers in these CCREs. Induced CCREs are enriched for the GR binding motif in both cell types (p=0.01 in A549, p=\num{1e-26} in U2OS, Table S3). The repressed CCREs in A549 cells are not significantly enriched in any TF motifs, while repressed CCREs in U2OS cells are enriched in motifs for homeobox factors, SMAD, CREM, and zinc factors (Table S3). GR could act on repressed CCREs via tethering to the locus by another DNA binding factor (\cite{weikum_glucocorticoid_2017}) or instead by an indirect mechanism. 
\clearpage


Approximately half of enhancers act on the nearest gene (\cite{fulco_activity-by-contact_2019}), and so we determined if the closest expressed gene to each dex-responsive CCRE is dex-responsive.  While not every closest gene was dex-responsive, nearest genes to induced CCREs have globally increased expression after dex treatment in both cell types (Figure \ref{fig:fig4}A, B), whereas repressed CCREs are repressed by dex (Figure \ref{fig:fig4}C, D). As with the repressed CCREs, the repressed nearest genes displayed little binding of GR at their promoters, implying further that their repression is likely an indirect consequence of GR signaling. 
\begin{figure}[hb]
\includegraphics[width=\textwidth]{figures/Figure4_v2_2021-04.png}
\caption{\textbf{Enhancer characteristics present in CCREs identified by nascent transcription. (A-D)} The closest annotated gene TSS for each differentially expressed CCRE was identified in A549 \textbf{(A, B)} and U2OS \textbf{(C, D)} cells. The expression changes for the associated genes are plotted for induced \textbf{(A, C)} and repressed \textbf{(B, D)} CCREs. Significance testing was performed with Wilcoxon rank sum tests; *p < 0.0005. \textbf{(E-H)} The closest induced \textbf{(E, G)} or repressed \textbf{(F, H)} CCRE to each gene was identified in A549 \textbf{(E, F)} and U2OS cells \textbf{(G, H)}, and the distances are plotted from the gene’s TSS. \textbf{(I)} The closest TSS of a differentially expressed gene was found for each differentially expressed CCRE in A549 cells. The expression changes for each gene and the nearest CCREs are plotted; n=48 genes. \textbf{(J)} The closest TSS of a differentially expressed gene was found for each differentially expressed CCRE in U2OS cells. The expression changes for each gene and the nearest CCREs are plotted; n=214 genes.}
\label{fig:fig4}
\end{figure}
We also determined that dex-responsive genes are located near dex-responsive CCREs. Because a gene’s expression can be influenced by multiple enhancers, we did not necessarily expect that the nearest CCRE would be dex-responsive. In both cell types, however, genes that are upregulated are closer to induced CCREs than genes that are downregulated or unaffected by dex treatment (Figure \ref{fig:fig4}E,G), while the converse is true for downregulated genes (Figure \ref{fig:fig4}F,H).  

We next examined if degree of transcriptional changes at dex-responsive CCREs correlates with expression changes of the nearest dex-responsive gene. Sometimes multiple CCREs had the same nearest gene, and so for those, we used the sum of the CCREs’ induction signal. In both cell types, a significant correlation exists between the degree of transcriptional change for CCREs and for genes after dex treatment (Figure \ref{fig:fig4}I,J). Interestingly, the slope of the best-fit line for gene and CCRE induction differ between the two cell types. In A549 cells (Figure \ref{fig:fig4}I), the slope is close to one, showing that CCREs and their neighboring genes are similarly induced. In U2OS cells (Figure \ref{fig:fig4}J), however, some CCREs have much stronger induction than their candidate target genes, implying that factors beyond transcription level affect enhancer activity on gene expression. Recent work shows that during macrophage differentiation, enhancers can act additively or synergistically (\cite{choi_evidence_2021}), which is consistent with our data. The overall correlation between CCRE transcription and nearby gene transcription strongly suggests that these CCREs enhance transcription of nearby target genes.

\subsection*{Transcription is a better measure of enhancer activity than GR binding}

\begin{figure}[ht]
\includegraphics[height=7.75in]{figures/Figure5_2021-04.png}
\caption{\textbf{Most GR-bound sites do not have increased transcription. (A)} UpSet plot showing overlap between GR binding and CCREs that are significantly induced (ind) or repressed (rep). (\textbf{B and C)} PRO-seq signal at GORs that have increased (“ind CCRE”) or no increase (“non-CCRE”) in transcription after 100 nM dex treatment in A549 \textbf{(B)} and U2OS \textbf{(C)} cells. \textbf{(D)} Cumulative distributions of distance between each GOR, based on the presence or absence of an induced CCRE, to the nearest induced gene. Significance testing by Wilcoxon rank sum tests; for both cell types p < \num{1e-15}. \textbf{(E)} Conservation at GORs with and without CCRE induction. \textbf{(F)} H3K27ac signal for GORs with and without CCRE induction.}
\label{fig:fig5}
\end{figure}

Strikingly, <5\% of GORs display significant changes in transcription in response to dex, whereas 67\% of dex-induced CCREs are GORs (Figure \ref{fig:fig5}A). Of the induced CCREs that lack GR binding, approximately 30\% reside within GR-inducible genes  and thus their apparent induction could be an artifact of the gene’s increased transcription. Others may be induced due to long-range interactions, GR tethering, or indirect effects of the dex treatment. Only 3\% of CCREs with repressed transcription are GR-bound (Figure \ref{fig:fig5}A). PRO-seq profiles of CCREs generally show a bidirectional pattern of transcription, with TF motifs between the two transcription start sites (\cite{core_analysis_2014, tippens_transcription_2020}). We examined the PRO-seq signal at GORs with and without induced CCREs in A549 and U2OS cells (Figure \ref{fig:fig5}B, C). These profiles are centered on the GR binding summit, and we observe the expected bidirectional pattern in induced CCREs, showing that although GR is not sufficient for transcription, its GR binding between the two transcripts likely causes induction. 

We were interested in determining if GORs with and without changing CCRE transcription differed in other ways. GORs with increased transcription were significantly closer to induced genes than ones without changed transcription (Figure 5D) and had higher conservation scores (Figure \ref{fig:fig5}E), demonstrating that they are under purifying selection. These sites have higher H3K27ac signal in control cells that sharply increases after 1 hour of dex treatment (Figure \ref{fig:fig5}F). H3K27ac is a chromatin modification that marks active enhancers (\cite{creyghton_histone_2010}). We considered that GORs without changing transcription at 45 minutes may gain enhancer activity later in the GR response. To address that hypothesis, we used existing H3K27ac data in A549 cells treated with dex for 12 hours (\cite{mcdowell_glucocorticoid_2018}). We found that after 12 hours, GORs with no change in transcription after 45 minutes of dex have less H3K27ac than ones with induced transcription (Figure \ref{fig:fig5}F). GORs with dex-responsive CCREs therefore have features that strongly correlate with enhancer activity. While GORs without changes in transcription may have roles in the cellular response to GR, they appear to have less direct enhancer activity than ones with induced transcription.

We reasoned that the GORs with induced transcription would differ in TF composition from those without induced transcription. GORs with induced transcription in both A549 and U2OS cells were selectively enriched for FOSL2 motifs, and A549-specific GORs were also enriched for Ets motifs (Table S3). Binding of the AP-1 factor FOSL2 is also found at GORs that are present in both A549 and U2OS cells (Figure \ref{fig:fig2}B), regardless of change in transcription. Of GORs bound in both cell types, 8\% have increased transcription, as compared to 1\% of A549-specific GORs and 2\% of U2OS-specific GORs (Figure \ref{fig:fig5}A). AP-1 motifs therefore likely potentiate the GR response. 

\clearpage

\subsection*{Pioneer factor binding distinguishes A549- and U2OS-specific CCRE induction}

We showed above that cell type-specific binding of GR corresponds to pioneer factor binding (Figure \ref{fig:fig2}). Here, we determined if pioneer factor binding relates to the transcriptional response at GORs by comparing CEBPB and FOXA2 binding in non-dex treated A549 cells (Figure \ref{fig:fig6}A, B). The loci at which GR binds specifically in U2OS cells show minimal binding of pioneer factors in A549 cells, whereas the intensity of CEBPB and FOXA2 binding is similar when comparing A549-specific GORs that do and do not overlap CCREs. These factors therefore appear to open chromatin and allow binding of other factors, but they are not sufficient for inducing transcription. As described above, AP-1 motifs mark the GORs that have induced transcription. In control A549 cells, the AP-1 component FOSL2 has greater binding to GORs that will have induced transcription upon dex addition (Figure \ref{fig:fig6}C), suggesting that AP-1 poises these GORs for GR induction. Induced GR binding then follows the same pattern as JUNB binding (Figure \ref{fig:fig6}D). We speculate from these findings that pioneer factors open chromatin, allowing AP-1 binding to mark the loci, which upon dex treatment enables GR binding and presumptive enhancer activity.

\begin{figure}[ht]
\includegraphics[width=\textwidth]{figures/Figure6_v2.png}
\caption{\textbf{Pioneer factors underlie cell type-specific GR induction.} ChIP-seq data for \textbf{(A)} CEBPB, \textbf{(B)} FOXA2, \textbf{(C)} FOSL2, and \textbf{(D)} GR from A549 cells are shown as metaplots centered on cell type-specific GORs of A549 and U2OS, with and without increased transcription.}
\label{fig:fig6}
\end{figure}

\section*{Discussion}
We demonstrate here that cell type-specific GR responses require non-GR transcription factors, some of which are broadly expressed, and some which have limited expression. These classes of TFs assemble in a combinatorial manner, and none alone is sufficient for the cellular response to glucocorticoids. We primarily focused on the non-promoter genomic loci at which GR binds, as these make up >95\% of GORs. Critically, we show that a small fraction of GORs has dex-responsive transcription in either cell type. Those responsive CCREs have binding sites and/or motifs for putative pioneer factors near the GR binding site, as well as for AP-1. Glucocorticoid-induced CCRE transcription therefore appears to require multiple classes of TFs, which have been implicated in other studies to affect distinct steps of the transcription cycle, including opening chromatin, maintaining accessibility, recruiting Pol II, and releasing Pol II into active elongation. Our study confirms and also expands upon previous work showing that a single stimulus acting on a single TF can regulate sets of genes that are the same in two cell types or specific to a particular cell type (\cite{gertz_distinct_2013, weikum_glucocorticoid_2017, vihervaara_molecular_2018})—likely a consequence of cell type-specific 3D chromatin architecture and of enhancers that are composed of different combinations of transcription factors. Importantly, it is likely that GR-mediated transcriptional regulation is conferred across the genome and in different cell types by a myriad of GR-containing transcriptional regulatory complexes, distinct in composition and conformation (\cite{weikum_glucocorticoid_2017}), and displaying different combinations of characteristics that have been correlated with enhancer function.
	
Previous studies of GR signaling demonstrated roles for pioneer factor binding (\cite{so_determinants_2007, belikov_foxa1_2009, grontved_cebp_2013, mcdowell_glucocorticoid_2018}), pre-wired enhancer-promoter loops (\cite{dippolito_pre-established_2018}), and AP-1 binding (\cite{so_determinants_2007, biddie_transcription_2011, vockley_direct_2016, mcdowell_glucocorticoid_2018}). Our critical addition was using PRO-seq to distinguish between GORs with and without nascent transcription, and to do so in two cell types that express different pioneer factors. GORs in A549 cells globally associate with the binding of FOXA2 and CEBPB pioneer factors (Figure \ref{fig:fig2}). A small fraction of those GORs exhibit differential transcription after dex treatment that are distinguished by also displaying AP-1 binding (Figure \ref{fig:fig6}). A subset of GORs is only present in U2OS cells. The corresponding loci in A549 cells lack FOXA2 and CEBPB binding and are less accessible, and so even though a GBS is present, the chromatin environment does not appear to be amenable to GR binding. Furthermore, AP-1 binding is seen at GR loci with dex-responsive CCRE transcription in A549 cells, and its motif is present at the transcribed GR loci in U2OS cells. AP-1 is therefore not essential for GR binding but may play a different role in regulating CCRE transcription. For example, AP-1 may maintain a larger region of accessible chromatin than FOXA or CEBP, thereby facilitating transcription at those sites. Alternatively, AP-1 may promote Pol II recruitment, poising the sites for transcription upon GR binding. GR has been reported to impact both Pol II recruitment and pausing at promoters (\cite{gupte_glucocorticoid_2013, sacta_gene-specific_2018}), and putative enhancers can also undergo Pol II pausing (\cite{core_analysis_2014, scruggs_bidirectional_2015, henriques_widespread_2018}). Thus our comparative analysis of GR activity in two cell types identifies different roles for cell type-specific factors and for the more broadly expressed AP-1 TF.

While only a small fraction of GORs displays induced transcriptional activity after 45 minutes of dexamethasone treatment, others may be transcribed at later time points. This mechanism could occur because GR may itself decompact chromatin and recruit AP-1 binding (\cite{vockley_direct_2016, jubb_glucocorticoid_2017}). Direct validation of GR-mediated transcriptional regulation by CCREs (\cite{ehmsen_definition_2019}), together with high-resolution and extensive time course measuring nascent RNA, TF and coactivator binding, and chromatin accessibility, will further illuminate how GR and other factors coordinate to regulate gene expression.

Our work raises an interesting question: what is the role of GR’s widespread binding in the cellular response to glucocorticoids? GORs that have changed transcription also include characteristics that correlate with enhancer activity. In particular, dex-responsive CCREs are close to dex-responsive genes. These sites also have higher conservation and increased H3K27ac after dex treatment. GR binds to thousands of other loci, however. Unequivocal determination of enhancer activity, and assessment of the role of transcription at those sites, would require extensive mutagenesis screening. We speculate that a subset of these GORs will undergo transcription later. Others could strengthen or modulate looping interactions in the genome and thus affect expression of genes distal from their enhancers. GR has also been implicated in condensate formation that increases the local concentration of transcription factors (\cite{garcia_intrinsically_2021}). Another possibility is that GORs that appear non-functional after dex treatment in A549 or U2OS cells play critical roles in other contexts, such as in response to a different glucocorticoid, at a specific developmental stage, or in a three-dimensional tissue architecture. Finally, because the genome is vast and DNA binding motifs are degenerate, it is likely that some GORs do not have a functional role.
	
The GR response has similarities to other inducible systems, including heat shock and TNF$\alpha$ treatment, showing that analysis of diverse systems of inducible gene expression can reveal common features of transcriptional regulation. In each of these instances, chromatin contacts between putative enhancers and promoters appear to be pre-wired (\cite{jin_high-resolution_2013, dippolito_pre-established_2018, ray_chromatin_2019}). The fact that three different systems that all include a rapid transcriptional response to an extracellular stimulus already have the required 3D chromatin architecture begs the questions of how these contacts are specified during development and to what other stressors cells are prepared to respond. In Drosophila, many promoter-enhancer loops form early in development before the gene is activated, and the promoters often have polymerase pausing (\cite{ghavi-helm_enhancer_2014}), which likely sets these genes up to be rapidly induced when needed. Further, GR, HSF1, and NF-$\kappa$B (the TF that directs the TNF$\alpha$ response) bind to many sites which fail to rapidly increase transcription in certain contexts (\cite{jin_high-resolution_2013, mahat_mammalian_2016, vockley_direct_2016, vihervaara_transcriptional_2017}). This TF binding without activity suggests that different contexts, including the presence and activity of other factors, are required for transcription. Both HSF1 and NF-$\kappa$B act by releasing paused Pol II (\cite{barboric_nf-b_2001, danko_signaling_2013, duarte_transcription_2016}), so the presence of other factors that facilitate Pol II recruitment and initiation may be essential (\cite{duarte_transcription_2016}). For GR, AP-1 binding may commonly play a role in GR-mediated induction of transcription, and more work is needed to define mechanisms by which AP-1 acts. These studies broadly show that transcription is regulated at multiple steps, all of which need to act in concert for the appropriate cellular response to hormone signaling.

\section*{Methods}
\paragraph{Cell culture.} U2OS and A549 human-derived parental cell lines were obtained from American Type Culture Collection and authenticated by short tandem repeat (STR) profiling (ATCC), and U2OS.hGR cells were generated previously in the Yamamoto laboratory (\cite{rogatsky_glucocorticoid_1997}). Cells were maintained in DMEM/low glucose (4 mM L-glutamine, 1000 mg/L glucose, 110 mg/L sodium pyruvate; HyClone Laboratories, Inc.) supplemented with 5\% FBS (Gemini BioProducts) at 37°C with 5\% CO2. 
\paragraph{PRO-seq library generation.} A549 and U2OS.hGR cells were treated with ethanol (Koptek) or 100 nM dexamethasone (Sigma) for 45 minutes. PRO-seq libraries were generated as described in (\cite{mahat_base-pair-resolution_2016}). Approximately \num{8e6} cells were collected by scraping in 1x PBS supplemented with 10 mM EDTA on ice. All spins were performed in a swinging bucket rotor at 4°C at 1000 g for 5 minutes. After pelleting, cells were then washed twice with 1x PBS. They were incubated in 5 mL permeabilization buffer (10 mM Tris-HCl, pH 7.4; 300 mM sucrose; 10 mM KCl; 5 mM MgCl2; 1 mM EGTA; 0.05\% Tween-20; 0.1\% NP40 substitute; 0.5 mM DTT, 1x Pierce Protease Inhibitor Tablet (ThermoFisher); 4 units per mL SUPERaseIn RNase Inhibitor (ThermoFisher)) for 5 minutes on ice, then washed twice in permeabilization buffer. Cells were resuspended in 100 $\mu$L storage buffer (50 mM Tris, pH 8.0; 25\% glycerol (v/v), 5 mM MgAc2, 0.1 mM EDTA, 5 mM DTT, 40 units per ml SUPERaseIn RNase Inhibitor (ThermoFisher)), then flash frozen in liquid nitrogen and stored at -80°C.

The run-on reaction mixture was made with biotinylated rCTP and rUTP (5 mM Tris, pH=8.0; 2.5 mM MgCl2; 0.5 mM DTT; 150 mM KCl; 200 $\mu$M rATP; 200 $\mu$M rGTP; 20 $\mu$M Biotin-11-rCTP (PerkinElmer); 20 $\mu$M Biotin-11-rUTP (PerkinElmer); 0.8 units per $\mu$L SUPERaseIn RNase Inhibitor (ThermoFisher); 50\% sarkosyl), then preheated at 37°C for at least 5 minutes. 100 $\mu$L of run-on mixture was added to 100 $\mu$L of thawed permeabilized cells, then incubated at 37°C for 5 minutes. The reaction was stopped by adding Trizol LS (ThermoFisher), followed by RNA isolation. RNA was fragmented on ice for 8 minutes with 0.2 N NaOH and neutralized with Tris pH 6.8, followed by buffer exchange with a Bio-Gel P-30 column (Bio-Rad).

Nascent RNA was enriched with M-280 Streptavidin Dynabeads (ThermoFisher), then Trizol extracted. In-line barcoded 3$\prime$ adapters were ligated with T4 RNA ligase I (NEB) for 12 hours at 20°C, followed by a second enrichment with M-280 Streptavidin Dynabeads (ThermoFisher) and Trizol extraction. The 5$\prime$ end was repaired by decapping with RppH (NEB) for 1 hour at 37°C, then phosphorylated with T4 PNK (NEB) for 1 hour at 37°C. The 5$\prime$ adapter, which included a 6-nucleotide UMI, was ligated with T4 RNA ligase I (NEB) for 12 hours at 20°C, followed by a third enrichment with M-280 Streptavidin Dynabeads (ThermoFisher) and Trizol extraction. cDNA was generated with SuperScript III (ThermoFisher), and libraries were amplified with Phusion for 13 cycles. Libraries were cleaned up with Ampure XP beads (Beckman Coulter) at a 1:1.6 ratio.

\paragraph{PRO-seq sequencing.} Libraries were sequenced on a HiSeq 4000 with single-end 50 nt reads. Libraries were separated based on their in-line barcodes with barcode\_splitter v0.18.0 (\cite{leach_barcode_2019}), PCR duplicates were removed with prinseq v0.20.4 (\cite{schmieder_quality_2011}), and adapters were trimmed with cutadapt v1.16 (\cite{martin_cutadapt_2011}). Reads were aligned to hg38 with bowtie2 v2.3.4.1 (\cite{langmead_fast_2012}), using end-to-end alignment and the sensitive settings. Uniquely mapping reads were subsampled to the sample with the lowest read number with samtools v1.7. Bedgraphs were generated in which the read numbers at the 3$\prime$ ends were reported using bedtools v2.26.0 (\cite{quinlan_bedtools:_2010}), then converted to bigwigs (\cite{kent_bigwig_2010}). Data were visualized in the Integrated Genome Viewer (\cite{robinson_integrative_2011}).

 \paragraph{PRO-seq data analysis.} Transcriptional regulatory elements (TREs) for each cell type and treatment were identified using dREG Gateway (\cite{danko_identification_2015, wang_identification_2019, chu_discovering_2019}) on subsampled data sets. Genes were considered potentially expressed if a TRE overlapped at least one annotated RefSeq transcription start site (TSS). TREs that were at least 1 kb away from an annotated (TSS) were considered candidate \emph{cis}-regulatory elements (CCREs), and a unified set of CCREs were determined by merging CCREs from each cell type and treatment, using bedtools v2.26.0 (\cite{quinlan_bedtools:_2010}).
 
 Read counts for genes and CCREs were found using the R v3.6 (\cite{r_core_team_r_2019}) package bigwig (\cite{martins_bigwig_2014}). Differential expression and log2 fold-change values were determined with DESeq2 (\cite{love_moderated_2014}), using a Benjamini-Hochberg-adjusted p-value cut-off of 0.05 as significantly differentially expressed. k-means clustering was performed with the PAM package in R, and differential expression data were plotted with the R packages ggplot2 (\cite{wickham_ggplot2_2016}) and pheatmap.
 
Motif enrichment was performed with HOMER v3.12 (\cite{heinz_simple_2010}) to find differentially enriched motifs, using 300 nt segments centered on the TRE, with all CCREs as the background set. The most enriched motifs were compared to HOCOMOCO v11 (\cite{kulakovskiy_hocomoco_2018}).

For analyses concerning expression of genes near CCREs, the gene set used was filtered to expressed genes, which were defined as having an annotated TSS overlapping a site of bidirectional transcription, as called from dREG. In Figure \ref{fig:fig3}B, all expressed genes were used, whereas in Figure \ref{fig:fig3}C, only ones that were significantly differentially expressed (Benjamini-Hochberg p-value < 0.05, log2 fold-change > 0) were included. The CCREs used in Figure 3C were required to be intergenic and within 100 kb of the closest dex-responsive gene. The TSS closest to each CCRE was found using bedtools closest (\cite{quinlan_bedtools:_2010}). 
Scripts used for these analyses are available at \url{https://github.com/ewissink/GR_enhancers}. 

\paragraph{NR3C1 ChIP-seq.} A549 and U2OS.hGR cells were treated for 90 minutes with dexamethasone (Sigma) at 100 nM or ethanol (Koptec), harvested by trypsinization, and fixed in suspension by adding 36.5\% formaldehyde to a final concentration of 1\% v/v; after incubating 3-10 minutes at room temperature (RT), formaldehyde was quenched by adding 2.5 M glycine to 0.3 M, RT 5 minutes, followed by transfer to ice. After washing twice in 20 mL ice-cold TBS (100 mM Tris-HCl, pH 7.5 @ 4°C/150 mM NaCl, centrifugation at 450g, 5 minutes, 4°C), cells were washed 3x in 1 mL RT MC lysis buffer (10 mM Tris-HCl, pH 7.5 @ RT, 10 mM NaCl, 3 mM MgCl2, 0.5\% (v/v) Tergitol type NP-40), resuspended in 1 mL RT MC lysis buffer, pelleted at 200g, 5 minutes, and frozen in liquid N2 for storage at -80°C.

Frozen nuclear pellets were resuspended in 180 $\mu$L MNase reaction buffer (10 mM Tris-HCl, pH 7.5 @ RT, 10 mM NaCl, 3 mM MgCl2, 1 mM CaCl2, 4\% (v/v) Tergitol type NP-40) supplemented with PMSF to 1:100, and the volume was adjusted to 270 $\mu$L with MNase reaction buffer. 1.35 $\mu$L MNase (New England Biolabs, diluted 1:10 in MNase reaction buffer) was added to the resuspended chromatin and incubated at 37°C, 5 minutes.  A cOmplete Mini, EDTA-free Protease Inhibitor Cocktail tablet (Roche) was dissolved in 500 $\mu$L MNase buffer (PIn cocktail); MNase reaction was stopped by adding 5.4 $\mu$L 0.2 M EGTA (pH 8), 7.2 µL 100 mM PMSF, 14.65 $\mu$L PIn cocktail, 14.65 µL 20\% SDS, and 14.4 $\mu$L 5 M NaCl, with gentle tube inversion to mix. 164 $\mu$L volume was transferred to each of two 1.5 mL Bioruptor® Plus TPX microtubes (Diagenode, Denville, NJ) and sonicated in a Bioruptor® Plus (UCD-300) Sonication System using intensity setting ‘H’ (320 W) and sonication parameters “CYCLE Num:30, Time ON:30sec, Time OFF:30sec”. During a 15-minute rest interval, samples were vortexed for 5 seconds, spun down in a microfuge, and transferred to new TPX tubes on ice, followed by a second round of sonication (amounting to 60 cycles total).

For GR ChIP, 150 $\mu$L Dynabeads™ Protein G slurry (Invitrogen) was mixed with 20 mg N499 (rabbit $\alpha$-human GR IgG) antibody (\cite{rogatsky_factor_2001}) plus 450 $\mu$L Lysis Buffer 2 (10 mM Tris-HCl, 1 mM EDTA, 150 mM NaCl, 5\% (v/v) glycerol, 0.1\% (w/v) sodium deoxycholate, 0.1\% (w/v) SDS, 1\% (v/v) Triton X-100, pH 8 @ 4°C; no PIn added) in a 1.5 mL Eppendorf tube, incubated 1 hour with rolling in 4°C cold room; tubes were placed in magnetic rack and supernatant was removed. Sonicated chromatin samples were pelleted at maximum speed, 10 minutes, 4°C, and white pellet and cloudy suspension above pellet were recovered by transferring from 1.5 $\mu$L TPX tubes to new 1.5 mL tubes. 10 $\mu$L aliquot of combined input was set aside for later processing. 25 $\mu$L 100X Halt™ Protease Inhibitor Cocktail (Thermo Scientific) was added to a 5 mL tube with 2.5 mL Dilution Buffer (identical to Lysis Buffer 2, except without SDS). ~275 $\mu$L chromatin (from approximately \num{1.5e7} cells) was added to the Dilution Buffer+beads in the 5 mL tube, effectively diluting the chromatin 1:10. The 5 mL tube was sealed with parafilm and incubated 4 hours on a roller in a 4°C cold room. During this time, input material was reverse-crosslinked by adding TE (10 mM Tris-HCl, 1 mM EDTA, pH 8; pH 7.5 @ RT) to a total volume of 80 $\mu$L, followed by addition of 100 $\mu$L ChIP Elution Buffer (50 mM Tris-HCl, pH 7.5 @ RT, 10 mM EDTA, 1\% SDS) and 20 $\mu$L Pronase (Roche, 20 mg/mL) for incubation at 42°C for 2 hours, then 65°C overnight.

100X Halt PIn was warmed to RT, and 15 $\mu$L was added to 1.5 mL of each of three Wash Buffers (A-C; Wash Buffer A: 10 mM Tris-HCl, 1 mM EDTA, 150 mM NaCl, 5\% (v/v) glycerol, 0.1\% (w/v) sodium deoxycholate, 0.1\% (w/v) SDS, 1\% (v/v) Triton X-100, pH 8.0 with Halt protease inhibitor mixture; Wash Buffer B: Buffer A with 500 mM NaCl and Halt protease inhibitor mixture; Wash Buffer C: 20 mM Tris-HCl, 1 mM EDTA, 250 mM LiCl, 0.5\% (v/v) Nonidet P-40, 0.5\% (w/v) sodium deoxycholate, Halt protease inhibitor mixture, pH 8.0). Beads were washed consecutively in 1.5 mL of each of the Wash Buffers A-C (1X Halt PIn), by gently adding buffer to resuspend the beads, then placing the tube on a magnetic rack and removing the supernatant after beads settled. Chromatin (from \num{1.5e7} cells on beads from 150 $\mu$L slurry) was eluted from beads in 300 $\mu$L Elution Buffer/Reverse-Crosslinking Buffer (10 mM Tris-HCl, 1 mM EDTA, 0.7\% (w/v) SDS, pH 8 @ RT) by incubating beads in buffer for 5 minutes, RT with gentle pipetting to occasionally mix, then allowing beads to settle on magnetic rack and transferring eluant volume to a new 1.5 mL Eppendorf tube. To reverse crosslinks, 450 $\mu$L Adjustment Buffer (50 mM Tris, 10 mM EDTA, 0.45\% SDS pH 7.0 at RT) was added with 82.5 $\mu$L Pronase (20 mg/mL) to 300 $\mu$L eluted chromatin, followed by incubation at 42°C for 2 hours, then 65°C overnight. DNA was subsequently cleaned from “input/MNase only” samples using a Qiagen PCR Purification Kit, and MinElute PCR Purification Kit (Qiagen) columns were used to purify ChIPs (one column/each ChIP, using 2.5 mL ERC buffer or 4.16 mL PB buffer), eluted in 15 µL EB. Recovered DNA was stored at -20°C. Libraries were generated using an Ovation® Ultralow System V2-32 (NuGEN Technologies, Redwood City, CA), quantified on a 2100 Bioanalyzer System (Agilent, Santa Clara, CA) with High Sensitivity DNA Kit. 

\paragraph{ChIP-seq sequencing.} GR ChIP-seq libraries were sequenced on a HiSeq 4000 with single-end 50 nt reads. The analysis was performed with Galaxy (\cite{afgan_galaxy_2018}). Reads were aligned to hg38 with bowtie v1.1.2 (\cite{langmead_ultrafast_2009}), using the setting --best and with one alignment reported per read. PCR duplicates were removed with samtools v1.3.1 rmdup. Peaks were called with MAC2 v2.1.1 (\cite{zhang_model-based_2008}), using the input as the control file. Bedgraphs of the q values were generated with MACS2 and used for downstream analyses. 

\paragraph{ChIP- and ATAC-seq analyses.} Fastq files for ChIP-seq and ATAC-seq datasets that were generated by other lab groups were downloaded from ENCODE or SRA, as described below. Reads were aligned to hg38 with bowtie2 (\cite{langmead_fast_2012}). Signal tracks were produced using MACS2 (\cite{zhang_model-based_2008}), using ENCODE pipelines (\cite{moore_expanded_2020}) (\url{https://github.com/ENCODE-DCC/atac-seq-pipeline}; \url{https://github.com/ENCODE-DCC/chip-seq-pipeline}). Data were visualized in the Integrated Genome Viewer (\cite{robinson_integrative_2011}). Heatmaps and metaplots were generated with Deeptools v3.4.3 (\cite{ramirez_deeptools2_2016}).

Additional datasets used. The following data sets were used in this study: U2OS.GR ATAC-seq, SRA: \href{https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE109589} (\cite{lee_different_2018}); A549 ATAC-seq, ENCODE: \href{https://www.encodeproject.org/experiments/ENCSR220ASC/}{ENCSR220ASC} (\url{https://www.encodeproject.org/}) (\cite{encode_project_consortium_integrated_2012, davis_encyclopedia_2018}); A549 FOXA2 ChIP-seq, SRA: \href{https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE90454}{GSE90454} (\cite{donaghey_genetic_2018}); A549 CEBPB ChIP-seq, ENCODE: \href{https://www.encodeproject.org/experiments/ENCSR701TCU/}{ENCSR701TCU}  (\cite{mcdowell_glucocorticoid_2018}); A549 H3K27ac ChIP-seq, ENCODE: \href{https://www.encodeproject.org/experiments/ENCSR778NQS/}{ENCSR778NQS} (\cite{mcdowell_glucocorticoid_2018}); A549 JUNB ChIP-seq, ENCODE: \href{https://www.encodeproject.org/experiments/ENCSR656VWZ/}{ENCSR656VWZ} (\cite{mcdowell_glucocorticoid_2018}), A549 FOSL2 ChIP-seq, ENCODE: \href{https://www.encodeproject.org/experiments/ENCSR593DGU/}{ENCSR593DGU} (\cite{mcdowell_glucocorticoid_2018}); A549 JUN ChIP-seq, ENCODE: \href{https://www.encodeproject.org/experiments/ENCSR192PBJ/}{ENCSR192PBJ} (\cite{mcdowell_glucocorticoid_2018}).

\paragraph{DATA ACCESS.} All raw and processed sequencing data generated in this study have been submitted to the NCBI Gene Expression Omnibus (GEO; \url{https://www.ncbi.nlm.nih.gov/geo/}) under accession numbers \href{GSE168767}{https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE168767} (PRO-seq) and \href{https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE163398}{GSE163398} (ChIP-seq).

\section*{Acknowledgments}
This work was supported by the NCI grant R01-CA020535 (to K.R.Y.), the NHGRI grant UM1-HG009393 (to J.T.L.), and the NIGMS grant F32-GM129904 (to E.M.W.). We thank the UCSF Center for Advanced Technology for high-throughput sequencing assistance. We also thank Charles Danko, the Lis lab, and Yamamoto lab for invaluable discussions. This LaTeX template was provided by George Kour. 

\printbibliography

\section*{Supplementary Material}
\paragraph{Table S1} Gene expression data from DESeq2
\paragraph{Table S2} MACS2 output for GR ChIP-seq
\paragraph{Table S3} HOMER output for motif searching
\paragraph{Table S4} CCRE expression data from DESeq2

\clearpage

\begin{figure}[ht!]
\includegraphics[width=\textwidth]{figures/FigureS1_2021-04.png}
\label{fig:figS1}
\end{figure}
}

\textbf{Figure S1, related to Figure 1. (A)} PCA of gene expression in A549 and U2OS cells after dex treatment. Cell types separate on the x-axis, dex dosages separate on the y-axis. \textbf{(B)} Spearman correlation coefficients for gene expression in A549 and U2OS cells after dex treatment from PRO-seq data. Two biological replicates (A and B) were performed for each treatment (EtOH, 1 nM dex, and 100 nM dex) in the two cell types. Replicates correlated well, whereas A549 and U2OS cells have very different expression profiles from each other. \textbf{(C)} Expression change of genes in A549 cells after 1 nM and 100 nM dex treatment. Genes with significant changes in expression have Benjamini-adjusted p < 0.05. Correlation between all genes is Spearman $\rho$ = 0.499, p<\num{1e-15}. \textbf{(D)} Expression change of genes in U2OS cells after 1 nM and 100 nM dex treatment. Genes with significant changes in expression have Benjamini-adjusted p < 0.05. Correlation between all genes is Spearman $\rho$ = 0.705, p<\num{1e-15}. 

\clearpage

\begin{figure}[ht!]
\includegraphics[width=\textwidth]{figures/FigureS2_2021-04.png}
\label{fig:figS2}
\end{figure}

\textbf{Figure S2, related to Figure 2. (A)} Gene expression differences from DESeq2 for transcription factors that have motif enrichment in A549 or U2OS cells (Table S1). FOXA1, FOXA2, CEBPA, and CEBPB are more highly expressed in A549 cells. RUNX1 and RUNX2 are more highly expressed in U2OS cells. NR3C1, which is the gene symbol for GR, is more highly expressed in A549 cells. FOS, FOSL1, FOSL2, JUN, and JUNB are all components of AP 1, and some are more highly expressed in A549 cells, while others are more highly expressed in U2OS cells. \textbf{(B)}  In A549 cells, metaplots of mean signal for JUN and JUNB in control cells are shown for shared GORs (blue), A549-specific GORs (green), and U2OS-specific GORs (pink). Signal is the -log10 p-value of enrichment, and shading indicates the standard deviation. \textbf{(C)} Number of GBS motifs present in GORs. *p=\num{2.2e-16}. \textbf{(D)} Enrichment of GR binding to promoters of induced and repressed genes in A549 and U2OS cells. Expected genes were calculated as (\# genes with GR promoter binding/total \# genes) * (\# differentially expressed genes). \textbf{(E)} Comparison of log2-fold change in expression for genes without and with GR binding at their promoters. * p < \num{1e-8}.
\clearpage

\begin{figure}[ht!]
\includegraphics[height=8in]{figures/FigureS3_2021-04.png}
\label{fig:figS3}
\end{figure}

\textbf{Figure S3, related to Figure 3.  (A)} UpSet plot showing genomic distribution of dREG-called bidirectional transcripts, which consist of promoters and CCREs (which can be intergenic or intragenic). Their cell-type specificity is also shown. \textbf{(B)} PCA of CCRE expression in A549 and U2OS cells after dex treatment. Cell types separate on the x-axis, dex dosages separate on the y-axis. \textbf{(C)} Spearman correlation coefficients for CCRE expression in A549 and U2OS cells after dex treatment from PRO-seq data. Two biological replicates (A and B) were performed for each treatment (EtOH, 1 nM dex, and 100 nM dex) in the two cell types. Replicates correlated well, whereas A549 and U2OS cells have very different CCRE expression profiles from each other. \textbf{(D)} CCREs with significant changes in expression have Benjamini-adjusted p < 0.05. Correlation between all CCREs is Spearman $\rho$ = -0.055, p<\num{1e-11}. (\textbf{E)} Expression change of CCREs in U2OS cells after 1 nM and 100 nM dex treatment. CCREs with significant changes in expression have Benjamini-adjusted p < 0.05. Correlation between all genes is Spearman $\rho$ = 0.440, p<\num{1e-15}.

\end{document}
